igvm/settings.py

```diff
@@ -193,6 +194,10 @@
     HypervisorAttributeValue('cpu_util_pct'),
     # Find Hypervisor with less I/O utilization
     HypervisorAttributeValue('iops_avg'),
+    # Find hypervisor with less used memory
+    HypervisorAttributeValue('libvirt_pool_used_gib'),
+    # Find hypervisor with less used disk
+    HypervisorAttributeValue('libvirt_memory_used_gib'),
```

Comment:
I wouldn't add these.
- They don't make sense if your pools aren't of the same size.
- The OtherVMs below probably still have more value then these.
- The HYPERVISOR_PREFERENCES mechanism is much to dumb to properly optimize placement anyway. You'd have to shift VMs around based on multiple attributes.

Reply:
That was ticket from you to sort HVs by free storage pool and memory. Unfortunately this function HypervisorAttributeValue sorts in way which makes sense for "used" and not "free" resource.

Reply:
It just doesn't make sense like this though and the InsufficientResource checks on top should be enough anyhow, if the fields cached from libvirt aren't stale. If they are stale, this won't help either.

Reply:
InsufficientResource also uses cached values and might still return HVs which are in fact not usable, even if we use values acquired in better way (from libvirt). It was never about optimal placement, it was about finding an usab le HV faster, as last check is done by locking a HV and getting real data from it.

Reply:
Let me frame it differently.
In the cases where we actually run this check we _might_ shave off between one and five hypervisor disk checks. Connecting to each and checking for the disks takes maybe .5 second. So best case we just saved 2.5 seconds. In a 5 to 15 minute igvm build/migrate. In the average run we probably save around 0 seconds.
On the downside we introduce code which makes little sense if your lvm pools aren't all of the same size.

Reply:
We have used graphite cached data attributes from serveradmin in igbalance before. The problem was exactly what you described above 

> TuxPowered42 15 days ago Author Member
> A freshly installed HV should be usable immediately, not after a few hours only when cron job on Serveradmin runs.

The information is outdated rather fast. So imagine you are evacuating a hypervisor or setup a fresh one. It becomes a very high rated candidate and will likely be on the top of the list. Problem is that after 1 or 2 migrations (or builds) the graphite information is outdated unless the graphite cache in serveradmin is updated. This may take some time as you already pointed out above. This hypervisor will be choosen over and over again and only get dismissed at the preflight check. The pre flight check is a very expensive operation requiring us to SSH to the hypervisor which is why we move it to the very end and try to execute it only if really needed.

This said I think you will not achieve what you expect. In single cases where we migrate or build single VMs or have a lot of hardware it will bring speed but for example for upcoming buster migration in areas with less hardware (Think of VN) and many migrations and builds in a row it will actully slow us down.

Reply:
@HerrSpace I lack real numbers but in my memory it is slower than half second per HV and there might be tens of HVs to check. For migrating of some small machines this adds up substantially.

Reply:
@kofrezo The information is outdated, that is true. But this moment we must check even more HVs, if we could skip just a few, we would gain some time. If you still don't agree then let's remove this part, work on Serveradmin cache speed and then reintroduce it.

Reply:
This part of change was removed from my PR.